spark/src/main/scala/org/apache/spark/sql/delta/stats/StatisticsCollection.scala

```diff
@@ -454,21 +454,32 @@ object StatisticsCollection extends DeltaCommand {
   }
 
   /**
-   * This method validates that the data type of data skipping column supports data skipping
-   * based on file statistics.
+   * This method validates that the data type of a data skipping column provided in
+   * DATA_SKIPPING_STATS_COLUMNS supports data skipping based on file statistics. If a struct
+   * column is specified, all its children are considered valid because stats will be collected
+   * for any supported types, and unsupported types will be ignored.
```

Comment:
1. 
```suggestion
   * for any supported types. For unsupported types, stats will not be collected.
```
@Kimahriman 

2. Could you please also add the reasons why we consider all data types specifically for the **struct** case to be considered valid here? E.g. to collect stats whenever we can and to be more lenient, as in if a child is unsupported then we will not through an exception and we have the stats of the other valid children.

3. If you can also specify the definition (can be short) of "valid" then it would also be awesome, I'm aware we didn't add it before but it would make things clearer.

Reply:
Made another update to try to clarify further